source/transactions/tests/legacy/mongos-recovery-token.yml

```diff
@@ -2,6 +2,8 @@ runOn:
     -
         minServerVersion: "4.1.8"
         topology: ["sharded"]
+        # serverless proxy doesn't use recovery tokens
+        serverless: "forbid"
```

Comment:
This made me think it may be worth testing a retry scenarios specific to serverless.

I thought it could be a risk that a driver assumes that responses to a sharded cluster always include the `recoveryToken` field. But I think the existing `tests/legacy/retryable-commit.yml` and `tests/legacy/retryable-writes.yml` should cover those cases.

Reply:
Yeah I think for now the coverage of that case is acceptable. Once it moves behind the LB and we have to write new tests anyways, that would probably be a good time to revisit this.